src/middlewared/middlewared/test/integration/assets/directory_service.py

```diff
@@ -191,3 +212,38 @@ def ldap(
         }
     finally:
         clear_ldap_info()
+
+
+def clear_ipa_info():
+    clear_ldap_info()
+    keytabs = call('kerberos.keytab.query')
+    for kt in keytabs:
+        call('kerberos.keytab.delete', kt['id'])
+
+
+@contextlib.contextmanager
+def ipa(
+    basedn=FREEIPA_BASEDN,
+    binddn=FREEIPA_ADMINDN,
+    bindpw=FREEIPA_ADMINPW,
+    hostname=FREEIPA_HOSTNAME,
+    nameserver=FREEIPA_IP,
+    **kwargs
+):
+    with override_nameservers(nameserver):
+        try:
+            config = call('ldap.update', {
+                "basedn": basedn,
+                "binddn": binddn,
+                "bindpw": bindpw,
+                "hostname": [hostname],
+                "ssl": "ON",
+                "auxiliary_parameters": "",
+                "validate_certificates": False,
+                "enable": True,
+                **kwargs
+            }, job=True)
+            del(config['bindpw'])
```

Comment:
flake8: missing whitespace after keyword.

And lots of other flake8 errors.

Reply:
Yeah, old flake8 issues, fixed anyway.